librz/core/canalysis.c

```diff
@@ -1005,25 +984,19 @@ static int __core_analysis_fcn(RzCore *core, ut64 at, ut64 from, int reftype, in
 				rz_flag_space_pop(core->flags);
 			}
 			rz_analysis_add_function(core->analysis, fcn);
-		}
-		if (fcn && has_next) {
-			ut64 newaddr = rz_analysis_function_max_addr(fcn);
-			RzIOMap *map = rz_io_map_get(core->io, newaddr);
-			if (!map || (map && (map->perm & RZ_PERM_X))) {
-				next = next_append(next, &nexti, newaddr);
-				for (i = 0; i < nexti; i++) {
-					if (!next[i]) {
-						continue;
-					}
-					rz_core_analysis_fcn(core, next[i], next[i], 0, depth - 1);
+			if (has_next) {
+				ut64 newaddr = rz_analysis_function_max_addr(fcn);
+				RzIOMap *map = rz_io_map_get(core->io, newaddr);
+				if (!map || (map && (map->perm & RZ_PERM_X))) {
+					next = newaddr;
+					rz_core_analysis_fcn(core, next, next, 0, depth - 1);
 				}
-				free(next);
+			}
+			if (core->analysis->cur && core->analysis->cur->arch && !strcmp(core->analysis->cur->arch, "x86")) {
```

Comment:
```suggestion
			if (core->analysis->cur && core->analysis->cur->arch && RZ_STR_EQ(core->analysis->cur->arch, "x86")) {
```

Reply:
I Haven't touched that line of code except moving it few lines up inside the big `if (fcn)`